### ISSUE TITLE
1.5.7 release has 6 test failures on Python 3.5

### ISSUE THREAD

**Issue by domenkozar** [Pylons/pyramid#1899]
Just noticed the tests failures, I'll take a look in following days what's going on.

```
======================================================================
FAIL: test_commit_conflict_with_two_includes (pyramid.tests.test_config.test_init.ConfiguratorTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/nix-build-python3.5-pyramid-1.5.7.drv-0/pyramid-1.5.7/pyramid/tests/test_config/test_init.py", line 1071, in test_commit_conflict_with_two_includes
    config.commit()
pyramid.exceptions.ConfigurationConflictError: Conflicting configuration actions
  For: ('view', None, '', None, 'd41d8cd98f00b204e9800998ecf8427e')
    Line 491 of file /tmp/nix-build-python3.5-pyramid-1.5.7.drv-0/pyramid-1.5.7/pyramid/util.py:
        return wrapped(self, *arg, **defaults)
    Line 491 of file /tmp/nix-build-python3.5-pyramid-1.5.7.drv-0/pyramid-1.5.7/pyramid/util.py:
        return wrapped(self, *arg, **defaults)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/tmp/nix-build-python3.5-pyramid-1.5.7.drv-0/pyramid-1.5.7/pyramid/tests/test_config/test_init.py", line 1074, in test_commit_conflict_with_two_includes
    self.assertEqual(c1, 'includeme1')
AssertionError: 'wrapper' != 'includeme1'
- wrapper
+ includeme1


======================================================================
FAIL: test_conflict_set_forbidden_view (pyramid.tests.test_config.test_init.ConfiguratorTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/nix-build-python3.5-pyramid-1.5.7.drv-0/pyramid-1.5.7/pyramid/tests/test_config/test_init.py", line 1130, in test_conflict_set_forbidden_view
    config.commit()
pyramid.exceptions.ConfigurationConflictError: Conflicting configuration actions
  For: ('view', <class 'pyramid.httpexceptions.HTTPForbidden'>, '', None, 'd41d8cd98f00b204e9800998ecf8427e')
    Line 491 of file /tmp/nix-build-python3.5-pyramid-1.5.7.drv-0/pyramid-1.5.7/pyramid/util.py:
        return wrapped(self, *arg, **defaults)
    Line 491 of file /tmp/nix-build-python3.5-pyramid-1.5.7.drv-0/pyramid-1.5.7/pyramid/util.py:
        return wrapped(self, *arg, **defaults)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/tmp/nix-build-python3.5-pyramid-1.5.7.drv-0/pyramid-1.5.7/pyramid/tests/test_config/test_init.py", line 1133, in test_conflict_set_forbidden_view
    self.assertEqual(c1, 'test_conflict_set_forbidden_view')
AssertionError: 'wrapper' != 'test_conflict_set_forbidden_view'
- wrapper
+ test_conflict_set_forbidden_view

======================================================================
FAIL: test_conflict_set_notfound_view (pyramid.tests.test_config.test_init.ConfiguratorTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/nix-build-python3.5-pyramid-1.5.7.drv-0/pyramid-1.5.7/pyramid/tests/test_config/test_init.py", line 1115, in test_conflict_set_notfound_view
    config.commit()
pyramid.exceptions.ConfigurationConflictError: Conflicting configuration actions
  For: ('view', <class 'pyramid.httpexceptions.HTTPNotFound'>, '', None, 'd41d8cd98f00b204e9800998ecf8427e')
    Line 491 of file /tmp/nix-build-python3.5-pyramid-1.5.7.drv-0/pyramid-1.5.7/pyramid/util.py:
        return wrapped(self, *arg, **defaults)
    Line 491 of file /tmp/nix-build-python3.5-pyramid-1.5.7.drv-0/pyramid-1.5.7/pyramid/util.py:
        return wrapped(self, *arg, **defaults)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/tmp/nix-build-python3.5-pyramid-1.5.7.drv-0/pyramid-1.5.7/pyramid/tests/test_config/test_init.py", line 1118, in test_conflict_set_notfound_view
    self.assertEqual(c1, 'test_conflict_set_notfound_view')
AssertionError: 'wrapper' != 'test_conflict_set_notfound_view'
- wrapper
+ test_conflict_set_notfound_view


======================================================================
FAIL: test_add_view_custom_predicate_bests_standard_predicate (pyramid.tests.test_config.test_views.TestViewsConfigurationMixin)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/nix-build-python3.5-pyramid-1.5.7.drv-0/pyramid-1.5.7/pyramid/tests/test_config/test_views.py", line 1471, in test_add_view_custom_predicate_bests_standard_predicate
    self.assertEqual(len(w), 1)
AssertionError: 3 != 1

======================================================================
FAIL: test_add_view_with_custom_predicates_match (pyramid.tests.test_config.test_views.TestViewsConfigurationMixin)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/nix-build-python3.5-pyramid-1.5.7.drv-0/pyramid-1.5.7/pyramid/tests/test_config/test_views.py", line 1435, in test_add_view_with_custom_predicates_match
    self.assertEqual(len(w), 1)
AssertionError: 2 != 1

======================================================================
FAIL: test_add_view_with_custom_predicates_nomatch (pyramid.tests.test_config.test_views.TestViewsConfigurationMixin)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/nix-build-python3.5-pyramid-1.5.7.drv-0/pyramid-1.5.7/pyramid/tests/test_config/test_views.py", line 1452, in test_add_view_with_custom_predicates_nomatch
    self.assertEqual(len(w), 1)
AssertionError: 2 != 1

```

**Comment by krysros** [comment]
... and 7 test failures on Python 3.5 on Windows.

    ======================================================================
    FAIL: test_route_static_views (pyramid.tests.test_scripts.test_proutes.TestPRoutesCommand)
    ----------------------------------------------------------------------
    Traceback (most recent call last):
      File "c:\users\kros\downloads\pyramid-1.5.7\pyramid-1.5.7\pyramid\tests\test_scripts\test_proutes.py", line 423, in test_route_static_views
        self.assertEqual(data, expected[index])
    AssertionError: Lists differ: ['__static2/', '/static2/*subpath', '/var/www/static\\', '*'] != ['__static2/', '/static2/*subpath', '/var/www/static/', '*']

    First differing element 2:
    /var/www/static\
    /var/www/static/

    - ['__static2/', '/static2/*subpath', '/var/www/static\\', '*']
    ?                                                     ^^

    + ['__static2/', '/static2/*subpath', '/var/www/static/', '*']
    ?                                                     ^

**Comment by domenkozar** [comment]
Closing as this is fixed on master

**Comment by bertjwregeer** [comment]
@domenkozar Not yet, this only fixes some of the issues, you are seeing some other test failures that we are not seeing on master.

Specifically:

```text
======================================================================
FAIL: test_add_view_custom_predicate_bests_standard_predicate (pyramid.tests.test_config.test_views.TestViewsConfigurationMixin)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/nix-build-python3.5-pyramid-1.5.7.drv-0/pyramid-1.5.7/pyramid/tests/test_config/test_views.py", line 1471, in test_add_view_custom_predicate_bests_standard_predicate
    self.assertEqual(len(w), 1)
AssertionError: 3 != 1

======================================================================
FAIL: test_add_view_with_custom_predicates_match (pyramid.tests.test_config.test_views.TestViewsConfigurationMixin)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/nix-build-python3.5-pyramid-1.5.7.drv-0/pyramid-1.5.7/pyramid/tests/test_config/test_views.py", line 1435, in test_add_view_with_custom_predicates_match
    self.assertEqual(len(w), 1)
AssertionError: 2 != 1

======================================================================
FAIL: test_add_view_with_custom_predicates_nomatch (pyramid.tests.test_config.test_views.TestViewsConfigurationMixin)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/nix-build-python3.5-pyramid-1.5.7.drv-0/pyramid-1.5.7/pyramid/tests/test_config/test_views.py", line 1452, in test_add_view_with_custom_predicates_nomatch
    self.assertEqual(len(w), 1)
AssertionError: 2 != 1
```

**Comment by bertjwregeer** [comment]
I'll take a look later today to see if I can replicate these issues, and if so, get a fix out for them.

**Comment by domenkozar** [comment]
:beers:

**Comment by warsaw** [comment]
Yep, we're seeing the last three build failures on Debian, where we have Python 3.5 enabled.

**Comment by warsaw** [comment]
Note, we're building 1.5.7 in Debian.  Against git tip, Python 3.5 all succeeds.  In 1.5.7 we see additional deprecation warnings:

```
======================================================================
FAIL: test_add_view_custom_predicate_bests_standard_predicate (pyramid.tests.test_config.test_views.TestViewsConfigurationMixin)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/python-pyramid-_xOhzM/python-pyramid-1.5.7+dfsg/pyramid/tests/test_config/test_views.py", line 1471, in test_add_view_custom_predicate_bests_standard_predicate
    self.assertEqual(len(w), 1, [str(wm) for wm in w])
AssertionError: 3 != 1 : ['{message : DeprecationWarning(\'The "custom_predicates" argument to Configurator.add_view is deprecated as of Pyramid 1.5.  Use "config.add_view_predicate" and use the registered view predicate as a predicate argument to add_view instead. See "Adding A Third Party View, Route, or Subscriber Predicate" in the "Hooks" chapter of the documentation for more information.\',), category : \'DeprecationWarning\', filename : \'/build/python-pyramid-_xOhzM/python-pyramid-1.5.7+dfsg/pyramid/tests/test_config/test_views.py\', lineno : 1468, line : None}', "{message : DeprecationWarning('inspect.getargspec() is deprecated, use inspect.signature() instead',), category : 'DeprecationWarning', filename : '/build/python-pyramid-_xOhzM/python-pyramid-1.5.7+dfsg/pyramid/config/util.py', lineno : 204, line : None}", "{message : DeprecationWarning('inspect.getargspec() is deprecated, use inspect.signature() instead',), category : 'DeprecationWarning', filename : '/build/python-pyramid-_xOhzM/python-pyramid-1.5.7+dfsg/pyramid/config/util.py', lineno : 204, line : None}"]

======================================================================
FAIL: test_add_view_with_custom_predicates_match (pyramid.tests.test_config.test_views.TestViewsConfigurationMixin)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/python-pyramid-_xOhzM/python-pyramid-1.5.7+dfsg/pyramid/tests/test_config/test_views.py", line 1435, in test_add_view_with_custom_predicates_match
    self.assertEqual(len(w), 1, [str(wm) for wm in w])
AssertionError: 2 != 1 : ['{message : DeprecationWarning(\'The "custom_predicates" argument to Configurator.add_view is deprecated as of Pyramid 1.5.  Use "config.add_view_predicate" and use the registered view predicate as a predicate argument to add_view instead. See "Adding A Third Party View, Route, or Subscriber Predicate" in the "Hooks" chapter of the documentation for more information.\',), category : \'DeprecationWarning\', filename : \'/build/python-pyramid-_xOhzM/python-pyramid-1.5.7+dfsg/pyramid/tests/test_config/test_views.py\', lineno : 1434, line : None}', "{message : DeprecationWarning('inspect.getargspec() is deprecated, use inspect.signature() instead',), category : 'DeprecationWarning', filename : '/build/python-pyramid-_xOhzM/python-pyramid-1.5.7+dfsg/pyramid/config/util.py', lineno : 204, line : None}"]

======================================================================
FAIL: test_add_view_with_custom_predicates_nomatch (pyramid.tests.test_config.test_views.TestViewsConfigurationMixin)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/python-pyramid-_xOhzM/python-pyramid-1.5.7+dfsg/pyramid/tests/test_config/test_views.py", line 1452, in test_add_view_with_custom_predicates_nomatch
    self.assertEqual(len(w), 1, [str(wm) for wm in w])
AssertionError: 2 != 1 : ['{message : DeprecationWarning(\'The "custom_predicates" argument to Configurator.add_view is deprecated as of Pyramid 1.5.  Use "config.add_view_predicate" and use the registered view predicate as a predicate argument to add_view instead. See "Adding A Third Party View, Route, or Subscriber Predicate" in the "Hooks" chapter of the documentation for more information.\',), category : \'DeprecationWarning\', filename : \'/build/python-pyramid-_xOhzM/python-pyramid-1.5.7+dfsg/pyramid/tests/test_config/test_views.py\', lineno : 1451, line : None}', "{message : DeprecationWarning('inspect.getargspec() is deprecated, use inspect.signature() instead',), category : 'DeprecationWarning', filename : '/build/python-pyramid-_xOhzM/python-pyramid-1.5.7+dfsg/pyramid/config/util.py', lineno : 204, line : None}"]
```

I can probably workaround these in Debian's 1.5.7

**Comment by warsaw** [comment]
inspect.getargspec() is deprecated in Python 3.5

**Comment by bertjwregeer** [comment]
@warsaw We just released Pyramid 1.6 today, and I just pushed some bugfixes up in #2210 for Pyramid 1.5.

**Comment by warsaw** [comment]
Perfect!  I just uploaded 1.6 to Debian.

**Comment by mmerickel** [comment]
Merged #2210.
